### PR TITLE
DRIVERS-1682: Convert GridFS spec tests to unified test format

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/DRIVERS-1682

Corresponding POC PR for PHPLIB: https://github.com/mongodb/mongo-php-library/pull/819

### CODE REVIEW

File: source/gridfs/tests/download.yml [hunk 1]
```diff
@@ -1,192 +1,241 @@
...
+          id: { $oid: "000000000000000000000005" }
+        expectError: { isError: true } # ChunkIsMissing
+  - description: "download when final chunk is missing"
+    operations:
+      - name: deleteOne
+        object: *bucket0_chunks_collection
+        arguments:
+          filter:
+            files_id: { $oid: "000000000000000000000005" }
+            n: 2
```
Comment (jmikola) [inline source/gridfs/tests/download.yml:183]: The [original test](https://github.com/mongodb/specifications/blob/926241af4090d4345057b966581fd329262bbc3b/source/gridfs/tests/download.yml#L147) used `n: 1`, which was incorrect copypasta from the preceeding [Download when an intermediate chunk is missing](https://github.com/mongodb/specifications/blob/926241af4090d4345057b966581fd329262bbc3b/source/gridfs/tests/download.yml#L128) test.

File: source/gridfs/tests/delete.yml [hunk 2]
```diff
@@ -1,123 +1,198 @@
...
+        documents:
+          - *file2_chunk0
+          - *file3_chunk0
+  - description: "delete when files entry does not exist"
+    operations:
+      - name: delete
+        object: *bucket0
+        arguments:
+          id: { $oid: "000000000000000000000000" }
+        expectError: { isError: true } # FileNotFound
```
Comment (jmikola) [inline source/gridfs/tests/delete.yml:156]: The original tests asserted specific error types that were never actually mentioned in the spec. I've retained their names as comments to preserve historical context, as was done in the [POC tests](https://github.com/mongodb/specifications/commit/e4808b84f804f8b08293dcd51ce107e10a40589f#diff-45d7492ffd2b7db6b2eb70da128598f1f743076e786f1a2dcebec47408ae6bd5R87).

I also created [DRIVERS-1684](https://jira.mongodb.org/browse/DRIVERS-1684) should we decide to address this down the line.

File: source/gridfs/tests/README.rst [hunk 3]
```diff
@@ -1,66 +1,37 @@
...
-In general, they will have the following sections:
+The YAML and JSON files in this directory are platform-independent tests
+meant to exercise a driver's implementation of GridFS. These tests utilize the
+`Unified Test Format <../../unified-test-format/unified-test-format.rst>`__.
 
-1. description
-2. arrange
-3. act
-4. assert
+Conventions for Expressing Binary Data
```
Comment (jmikola) [inline source/gridfs/tests/README.rst:16]: This section was written with [this comment from SPEC-1216](https://jira.mongodb.org/browse/SPEC-1216?focusedCommentId=2142974&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2142974) in mind.

File: source/gridfs/tests/README.rst [hunk 4]
```diff
@@ -1,66 +1,37 @@
+============
```
Comment (jmikola) [inline source/gridfs/tests/README.rst:1]: Let me know if this file needs a changelog. I didn't think it was necessary as it was basically rewritten from scratch, but perhaps we'd like it in place for future use.

Comment (alcaeus) [reply source/gridfs/tests/README.rst:1]: I don't see any advantage with an empty changelog, so feel free to omit it.

**Review by alcaeus — APPROVED** [review]

**Pull request merged**